### PR TITLE
add kubernetes deploy support

### PR DESCRIPTION
**Pull request by jepp3**

### CODE REVIEW

File: Jenkinsfile-deploy [hunk 1]
```diff
@@ -0,0 +1,20 @@
+library 'whatsout'
+
+node('deploy-kubernetes'){
```
Comment (jepp3) [inline Jenkinsfile-deploy]: used for internal deploy with jenkins

**Comment by jepp3** [comment]
maven will create docker images and push them to dockerhub. For now i use my personal dockerhub account, i cannot see that our company has an account. But in the future we might want to change that. 

There is one deployment with two containers. The ingress will route "api" to backend and "app" to frontend. 

The caching issue is scoped out from this PR.

File: impl/pom.xml [hunk 2]
```diff
@@ -104,4 +104,61 @@
             <artifactId>jsonevent-layout</artifactId>
         </dependency>
     </dependencies>
+
+    <build>
```
Comment (jepp3) [inline impl/pom.xml]: In both ui and impl we have defined this plugin that during install creates docker images and during deploy pushes them to dockerhub.

File: impl/config.yml.tmpl [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+logging:
+    additivity:
+        com.fortnox.reactivewizard.ReactiveWizardServer: false
+        se.fortnox.reactivewizard.jaxrs.JaxRsResource: false
+    levels:
+        com.fortnox.reactivewizard.ReactiveWizardServer: INFO
+        se.fortnox.reactivewizard.jaxrs.JaxRsResource: INFO
+    appenders:
+        stdout:
+            layout: net.logstash.log4j.JSONEventLayoutV1
```
Comment (splitfeed) [inline impl/config.yml.tmpl]: Why use JSON output in the local dev config?

File: impl/pom.xml [hunk 4]
```diff
@@ -104,4 +104,61 @@
...
+                </executions>
+            </plugin>
+            <plugin>
+                <groupId>io.fabric8</groupId>
+                <artifactId>docker-maven-plugin</artifactId>
+                <extensions>true</extensions>
+                <configuration>
+                    <images>
+                        <image>
+                            <name>docker.io/jesperlahdevirta/rocket-fuel:${project.version}</name>
```
Comment (splitfeed) [inline impl/pom.xml]: jesperlahdevirta? :)

File: Jenkinsfile-deploy (hunk 1 above)
Comment (splitfeed) [reply Jenkinsfile-deploy]: It feels weird to have our specific in-house deployment code in the open source repo where no one else will be able to use it anyway. I guess it's not harmful, but it increases complexity for everyone else without providing value.

Same for config.json and the kube-folder.

Comment (jepp3) [reply Jenkinsfile-deploy]: The kube folder is universal. Its only the config.json that is not. The complexity increased by moving it to another repository is even higher.

File: impl/config.yml.tmpl (hunk 3 above)
Comment (jepp3) [reply impl/config.yml.tmpl]: Its a config that is included in the docker image.

File: impl/pom.xml (hunk 4 above)
Comment (jepp3) [reply impl/pom.xml]: Will be fixed, when we reach outside of MVP.

File: Jenkinsfile-deploy (hunk 1 above)
Comment (jepp3) [reply Jenkinsfile-deploy]: But i agree, its not fun to have it here.

**Pull request merged**